routers/api/v2/users.go

```diff
@@ -280,6 +281,99 @@ func (r *apiV2Router) RemoveFromTeam(ctx *gin.Context) {
 	ctx.Status(http.StatusNoContent)
 }
 
+// PUT: /api/v2/users/(:id|me)/password
+// x-www-form-urlencoded
+// Request:  password string
+// Response:
+// Headers:  Authorization -> token
+func (r *apiV2Router) SetPassword(ctx *gin.Context) {
+	var req struct {
+		Password string `form:"password"`
+	}
+	_ = ctx.Bind(&req)
+	if len(req.Password) == 0 {
+		r.logger.Debug("password not specified", zap.Int("password length", len(req.Password)))
+		models.SendAPIError(ctx, http.StatusBadRequest, "request must include the new password")
+		return
+	}
+
+	userId := ctx.Param("id")
+	if userId == "me" {
+		userIdObj, err := r.authorizer.GetUserIdFromToken(r.GetAuthToken(ctx))
+		if err != nil {
+			switch errors.Cause(err) {
+			case v2.ErrInvalidToken:
+				r.logger.Debug("invalid token", zap.Error(err))
+				r.HandleUnauthorized(ctx)
+			case v2.ErrInvalidTokenType:
+				r.logger.Debug("invalid token type", zap.Error(err))
+				models.SendAPIError(ctx, http.StatusBadRequest, "provided token is of invalid type for the requested operation")
+			default:
+				r.logger.Error("could not extract token type", zap.Error(err))
+				models.SendAPIError(ctx, http.StatusInternalServerError, "something went wrong")
+			}
+			return
+		}
+		userId = userIdObj.Hex()
+	}
+
+	err := r.userService.UpdateUserWithID(ctx, userId, services.UserUpdateParams{
+		entities.UserPassword: req.Password,
+	})
```

Comment:
I just noticed.....This is not very secure 🤣

Reply:
The password needs to be hashed before we pass it to the `UpdateUserWithID` method

Reply:
Fixed 😅